protocol/op-stack-manager-arch.md

```diff
@@ -0,0 +1,292 @@
+# Purpose
+
+OP Stack Manager is a contract that will deploy the L1 contracts for standard OP Stack chains in a single transaction.
+However, we also have use cases for deploying non-standard chains, such as deploying devnet and testnet contracts while developing interop.
+This document will outline an architecture that:
+
+- Enables the OP Stack Manager to deploy these non-standard chains as an initial milestone.
+- Gets us most of the way towards supporting standard chains deployments for the subsequent milestone.
+
+# Summary
+
+The original OP Stack Manager (OPSM) milestones were:
+
+- Milestone 1: Eliminating key handover complexity.
+- Milestone 2: Simplifying Superchain upgrades.
+- Milestone 3: Supporting interop upgrades.
+
+This design doc proposes the architecture for a Milestone 0.5 of "OPSM is used for all interop devnet & testnet L1 deployments".
+Adding this milestone will allow the interop team to build testing infrastructure that wraps OPSM, unifying our development, testnet, and production testing and deployment processes.
+
+# Problem Statement + Context
+
+*Some text copied from https://github.com/ethereum-optimism/design-docs/pull/52, which is related to this design doc.*
+
+The current L2 chain deployment approach originates from a time with Hardhat, single L1 target, and a single monolithic set of features.
+Since then the system has migrated to Foundry and extended for more features, but remains centered around a single monolithic deploy-config for all its features.
+
+The interop team needs a way to configure new multi-L2 deployments: The number of ways to compose L2s in tests grows past what a single legacy config template can support.
+
+Outside of interop, deployment also seems increasingly complex and opaque, while it does not have to be, due to the same configuration and composability troubles.
+
+Part of the solution to this is described in https://github.com/ethereum-optimism/design-docs/pull/52,and the other part involves restructuring our Solidity config processing and deploy scripts.
+This design doc focuses on the L1 contract configuration and deployment part of the solution.
+
+# Alternatives Considered
+
+The primary alternative is for the interop team to not use OPSM for devnet and testnet deployments.
+This is not ideal for various reasons:
+
+- Duplication of effort: The interop team would need to develop and maintain their own deployment scripts, even though new ones will already be written for OPSM.
+- Divergence: With duplication, it's likely the interop team's deployment scripts diverge from the production deployment scripts, leading to inconsistencies and bugs.
+
+# Proposed Solution
+
+There are two steps to deploying L1 contracts:
```

Comment:
It lists three steps. The superchain-contracts and shared implementation contracts can (and should) be bundled (at high level at least, implementation can do it step-wise).

Reply:
I thought about whether or not those two steps should be bundled. I went with separating them out because:
- We'll often need to deploy new implementation contracts (for upgrades) independently from new superchain contracts
- You could have different superchains that use the same implementation contracts, though I don't think that's likely

I've fixed the text to say three steps in fb6bbb6